stglib/sg/sgutils.py

```diff
@@ -336,3 +124,203 @@ def read_hex(filnam):
             hexmeta["TemperatureCalibrationTA3"] = float(col[3])
     f.close()
     return hexmeta
+
+
+def read_wb(filnam, encoding="utf-8"):
+    """Read data from an SBE 26plus Seagauge .wb file into an xarray dataset."""
+
+    burst_no = []
+    start_time = []
+    values = []
+    pressure = []
+
+    with open(filnam) as file:
+        for line in file:
+            if "SBE" in line:
+                continue
+            elif "*" in line:
+                col = line.split()
+                burst_no.append(int(col[1]))
+                start_time.append(int(col[2]))
+                sample_no = int(col[4])
+                sample_rows = math.floor(
+                    sample_no / 4
+                )  # Take total number of samples in each burst and divide by 4 columns to get number of rows to iterate
+
+                # Loop through rows for each burst number
+                for j in range(sample_rows):
+
+                    row = file.readline().rstrip()
+                    col = row.split()
+                    values.append(col)
+
+                # Convert to numpy float and reshape to 1 row
+                burst = np.float64(values)
+                burst = np.reshape(burst, (1, -1))
+
+                # Convert back to list to append bursts
+                burst = burst.tolist()
+                pressure.append(burst)
+
+                # Clear values for next burst
+                values = []
+    file.close()
+
+    # Convert pressure to numpy array
+    pressure = np.float64(pressure)
+    pressure = pressure.squeeze()
+
+    # Convert time
+    start_time = int_to_date(np.array(start_time))
+
+    # Create sample variable
+    sample = list(range(1, sample_no + 1))
+
+    # Make xarray
+    sg = xr.Dataset(
+        data_vars=dict(
+            burst_number=(["time"], burst_no),
+            sample=(["sample"], sample),
+            P_1=(["time", "sample"], pressure),
+        ),
+        coords=dict(time=start_time),
+    )
+
+    return sg
+
+
+def int_to_date(int_time):
+    """Read integer time in seconds since January 1, 2000 and convert to datetime"""
+
+    dt = pd.Timestamp("2000-01-01T00:00:00") + pd.to_timedelta(int_time, unit="s")
+
+    # This gave me a nanosecond precision error
+    # t0 = np.datetime64("2000-01-01T00:00:00")
+    # dt = t0 + int_time.astype("timedelta64[s]")
+
+    return dt
+
+
+def atmos_correct_burst(ds, atmpres):
+    met = xr.load_dataset(atmpres)
+    pressure = []
+
+    # Apply the correction for each burst in turn
+    ds["P_1ac"] = xr.full_like(ds["P_1"], np.nan)
+    for burst in ds.burst_number:
```

Comment:
This type of looping is often not necessary with coordinates associated with data in an xarray Dataset. Let's talk about whether there is a more straightforward way to do this.

Reply:
I talked to Steve about how the atmospheric pressure correction is done here.  I was applying one correction for each burst (which was 2.5 hours long).  He wants me to subdivide the bursts into our smaller calculated wave intervals before I do the pressure correction, so I am going to change this whole section of code.